### PR TITLE
Make device settings language have more impact.

### PR DESCRIPTION
**Pull request by rtibbles**
### Summary
Moves device settings language up in the priority list so that it takes priority over browser accept headers if set.

### Reviewer guidance
If you have no set session language, and no language cookie set, does it default to the device setting language, and not your browser language setting?

----

### Contributor Checklist

PR process:

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [x] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/core/device/translation.py [hunk 1]
```diff
@@ -13,7 +14,8 @@
 
 from .models import DeviceSettings
 
-DEVICE_LANGUAGE = None
```
Comment (jonboiser) [inline kolibri/core/device/translation.py:16]: the `global DEVICE_LANGUAGE` statement on line 29 can be removed too.

**Comment by jonboiser** [comment]
Tested this out by updating the database and it seems to work. I had to add a line to clear the cache on every refresh to avoid waiting 10 minutes. 

1. Changing DB doesn't change if user has set a language cookie
1. changing db does change language if in incognito
1. When language is null, it defaults to english for me

When implementing #2258, we will probably have to clear or re-set the cached language setting when we change the language to make sure that the changes take place immediately after a user updates them.

**Review by jonboiser — APPROVED** [review]

**Comment by rtibbles** [comment]
The tricky thing with the cache clearing is that our caches are thread safe, and so when we are running in production mode, the cache only gets cleared in the thread where the change is made. The only way to be sure of making the change is never to cache the device setting value.

**Comment by jonboiser** [comment]
Caching the device setting seems like a pretty minor optimization, I think, so maybe 
 we don't need to cache. The get-language function only gets called once when a user loads the app.

**Comment by rtibbles** [comment]
The get language from request gets called on every request (including API requests), until a user explicitly sets a language using the language switcher, when it gets set on the cookie and the session, and hence the read from the device setting gets by passed.

I think we might still be able to get away with not caching it (in the follow up PR #4994 - which uses language URLs, and so bypasses the need to check the device setting).

**Pull request merged**